framework/src/modules/chain/dpos/delegates_info.js

```diff
@@ -85,6 +89,16 @@ class DelegatesInfo {
 
 		// Perform updates that only happens in the end of the round
 		if (this._isLastBlockOfTheRound(block)) {
+			if (undo) {
```

Comment:
Does it maybe make more sense to have another function. Something like: `updateDelegateListAtTheEndOfRound`. This function, when called with undo can delete the delegate list, otherwise, in case of apply, it will create the delegate list for the new round.

Reply:
I agree with this also your other concern if we provide a future round number to getRoundDelegates method make sense. I'll create a separate issue to tackle that.https://github.com/LiskHQ/lisk-sdk/issues/4218